diff --git a/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/presentation/builders/ExpoNotificationBuilder.kt b/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/presentation/builders/ExpoNotificationBuilder.kt
index 610d303..725f2ee 100644
--- a/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/presentation/builders/ExpoNotificationBuilder.kt
+++ b/node_modules/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/presentation/builders/ExpoNotificationBuilder.kt
@@ -112,6 +112,11 @@ open class ExpoNotificationBuilder(
     notificationContent.badgeCount?.toInt()?.let { builder.setNumber(it) }
     notificationContent.categoryId?.let { addActionsToBuilder(builder, it) }
 
+    // Enable looping sound for alarms
+    if (notificationContent.categoryId == "alarm-actions") {
+      builder.setInsistent(true)
+    }
+
     applySoundsAndVibrations(content, builder)
 
     if (content.body != null) {
